packages/ia-components/sandbox/theatres/with-youtube-spotify/audio-with-youtube-spotify-main.jsx

```diff
@@ -177,6 +185,30 @@ class AudioPlayerWithYoutubeSpotify extends Component {
     return audioSource || {};
   }
 
+  /**
+   * Finds the track to highlight for the track listing
+   *
+   * @param { object } audioSource
+   * @param audioSource.index - JWPlayer audio source
+   * @param audioSource.trackNumber - Third party audio source
+   */
+  getTrackToHighlight(audioSource) {
+    const { trackSelected, trackStartingPoint } = this.state;
+    if (Number.isInteger(audioSource.index)) {
```

Comment:
How come you have to check for `isInteger` for all of these checks? Or are you checking for existence?

Reply:
existence & type check